checker/tests/nullness/java8inference/Issue1032.java

```diff
@@ -16,7 +16,7 @@ public class Issue1032 {
         return arg.map(Issue1032::castStringToNonNull);
     }
 
-    @SuppressWarnings("nullness")
+    @SuppressWarnings("nullness", "cast.unsafe")
     static <T> @NonNull T castTToNonNull(@Nullable T arg) {
```

Comment:
Casting an expression of type variable `T` to `T`, we now have a more fine-grained approach.
We'll compare the lower bound of cast type with the upper bound of the expression type. If the former is the supertype of the latter, we are sure that if must an upcast. Otherwise, it cannot be statically verified.